src/TraceEvent/TraceLog.cs

```diff
@@ -1106,13 +1108,11 @@ unsafe private void SetupCallbacks(TraceEventDispatcher rawEvents)
                     moduleFile.productName = lastFileVersionData.ProductName;
                 }
 
-                /* allow these to remain in the trace.  Otherwise you can't just look at the events view and look up DLL info
-                 * which is pretty convenient.   If we have a good image view that we can remove these. 
-                if (data.Opcode == TraceEventOpcode.DataCollectionStart)
-                    bookKeepingEvent = true;
-                else if (data.Opcode == TraceEventOpcode.DataCollectionStop)
-                    bookKeepingEvent = true;
-                 ***/
+                // Remember this ModuleFile because there can be Image* events after this with 
+                // the same timestamp that have information that we need to put  into it 
+                // (the logic above handles the case when those other events are first).  
+                lastTraceModuleFile = moduleFile;
+                lastTraceModuleFileQPC = data.TimeStampQPC;
```

Comment:
Yesterday Aaron had mentioned that the QPC timestamps might not be exact. I'm not sure exactly in what scenario that happens though.

Reply:
Hmm, that is surprising, can we get someone with more details?   These events are injected after the fact, so it would be really surprising that they were not identical (they HAVE To put some value there, it is hard to believe it would not be the clone of the event they are trying to annotate).   

More generally, I need to know what is the 'right' way of 'knowing' that these two events are meant to be associated.